src/DynamoCoreWpf/ViewModels/Search/NodeAutoCompleteSearchViewModel.cs

```diff
@@ -155,98 +164,296 @@ private void InitializeDefaultAutoCompleteCandidates()
-            // Case 1: no results (0 items)
-            FilteredResults = new List<NodeSearchElementViewModel>();
-            DisplayNoRecommendationsLowConfidence = true;
-            DisplayLowConfidence = false;
-            NoRecommendationsOrLowConfidenceTitle = Resources.AutocompleteNoRecommendationsTitle;
-            NoRecommendationsOrLowConfidenceMessage = Resources.AutocompleteNoRecommendationsMessage;
+            // Intialize request for the the ML API
+            MLNodeAutoCompletionRequest request = new MLNodeAutoCompletionRequest();
+            DisplayNoRecommendationsLowConfidence = false;
+
+            var nodeInfo = PortViewModel.PortModel.Owner;
+            var portInfo = PortViewModel.PortModel;
+
+            request.DynamoVersion = AssemblyHelper.GetDynamoVersion().ToString();
+            request.NumberOfResults = 10;
 
-            // Case 2: the result has at least one item assuming each node could be by recommendation or by use
-            /*
-            FilteredResults = DefaultResults.Where(e => e.Name == "Watch 3D" || e.Name == "Python Script").ToList();
-            foreach (var item in FilteredResults)
+            // Set node info
+            request.Node.Id = nodeInfo.GUID.ToString();
+            request.Node.Lacing = nodeInfo.ArgumentLacing.ToString();
+
+            if (nodeInfo is DSFunctionBase functionNode)
             {
-                item.Model.AutoCompletionNodeMachineLearningInfo.ViewConfidenceScoreRecentUse = true;
-                item.Model.AutoCompletionNodeMachineLearningInfo.IsByUse = true;
+                request.Node.Type.Id = functionNode.CreationName;
             }
-            DisplayNoRecommendationsLowConfidence = !FilteredResults.Where(n => n.Model.AutoCompletionNodeMachineLearningInfo.IsByRecommendation).Any();
-            DisplayLowConfidence = false;
-            NoRecommendationsOrLowConfidenceTitle = Resources.AutocompleteNoRecommendationsTitle;
-            NoRecommendationsOrLowConfidenceMessage = Resources.AutocompleteNoRecommendationsMessage;
-            */
+            else if (nodeInfo is NodeModel nodeModel)
+            {
+                string typeId = GenerateTypeIdForNodeModelNode(nodeModel.GetType().AssemblyQualifiedName);
+                request.Node.Type.Id = typeId;
+            }
+
+            // Set port info
+            // If the node is a Variable-input nodemodel or zero-touch node, then set the port name as empty.
+            request.Port.Name = (nodeInfo is VariableInputNode || nodeInfo is DSVarArgFunction) ? string.Empty : portInfo.Name;
+            request.Port.Direction = portInfo.PortType == PortType.Input ? PortType.Input.ToString().ToLower() : PortType.Output.ToString().ToLower();
+            request.Port.KeepListStructure = portInfo.KeepListStructure.ToString();
+            request.Port.ListAtLevel = portInfo.Level;
 
+            // Set host info
+            request.Host.Name = dynamoViewModel.Model.HostAnalyticsInfo.HostName;
```

Comment:
I think you need to check if this is null and fallback to `dynamoViewModel.Model.HostName` or you need to make `HostAnalyticsInfo .HostName` fallback itself. (this causes issues for CER as well)